crates/router/src/workflows/outgoing_webhook_retry.rs

```diff
@@ -34,202 +38,379 @@ use crate::{
+pub async fn trigger_outgoing_webhook_bulk_workflow(
+    state: &SessionState,
+    process: &storage::ProcessTracker,
+) -> Result<(), errors::ProcessTrackerError> {
+    let tracking_data: OutgoingWebhookTrackingData = process
+        .tracking_data
+        .clone()
+        .parse_value("OutgoingWebhookTrackingData")?;
+    let merchant_id = tracking_data.merchant_id.clone();
+    let profile_id = tracking_data.business_profile_id.clone();
+    let primary_object_id = tracking_data.primary_object_id.clone();
+    let primary_object_created_at = tracking_data.primary_object_created_at;
+    let event_type = tracking_data.event_type;
+    let event_class = tracking_data.event_class;
+    let primary_object_type = tracking_data.primary_object_type;
+    let db = &*state.store;
+    let key_manager_state = &state.into();
+
+    let merchant_key_store = db
+        .get_merchant_key_store_by_merchant_id(
+            key_manager_state,
+            &merchant_id,
+            &db.get_master_key().to_vec().into(),
+        )
+        .await?;
+    let merchant_account = db
+        .find_merchant_account_by_merchant_id(key_manager_state, &merchant_id, &merchant_key_store)
+        .await
+        .to_not_found_response(errors::ApiErrorResponse::MerchantAccountNotFound)?;
+
+    let business_profile = utils::validate_and_get_business_profile(
+        db,
+        key_manager_state,
+        &merchant_key_store,
+        Some(&profile_id),
+        &merchant_id,
+    )
+    .await?
+    .ok_or(errors::ApiErrorResponse::ProfileNotFound {
+        id: profile_id.get_string_repr().to_string(),
+    })?;
+
+    let cloned_state = state.clone();
+    let (content, _) = Box::pin(get_outgoing_webhook_content_and_event_type(
+        state.clone(),
+        state.get_req_state(),
+        merchant_account.clone(),
+        merchant_key_store.clone(),
+        &tracking_data,
+    ))
+    .await?;
+    let webhook_details = webhooks_core::get_webhook_details_for_event_type(
+        &business_profile,
+        event_type,
+    )
+    .map_err(|err| {
+        logger::error!(original_webhook_error=?err, business_profile_id=?business_profile.get_id(), "Failed to get webhook details for event type");
+        errors::ProcessTrackerError::ResourceFetchingFailed {
+            resource_name: "webhook_details".to_string(),
+        }
+    })?;
+
+    if webhook_details.is_empty() {
+        logger::debug!(
+            business_profile_id = ?business_profile.get_id(),
+            event_type = ?event_type,
+            "No active webhook details found for event type or list is empty"
+        );
+        return Ok(());
+    }
+
+    let key_store = db
+        .get_merchant_key_store_by_merchant_id(
+            key_manager_state,
+            &tracking_data.merchant_id,
+            &db.get_master_key().to_vec().into(),
+        )
+        .await?;
+    for webhook_detail in webhook_details.iter() {
+        let webhook_detail_clone = webhook_detail.clone();
```

Comment:
Nit: You should be able to do something like:


```suggestion
    for webhook_detail in webhook_details {
```

And you can avoid the `webhook_detail.clone()`.